packages/protocol/contracts/L2/TaikoL2.sol

```diff
@@ -104,13 +104,12 @@ contract TaikoL2 is AddressResolver, ReentrancyGuard, IHeaderSync {
         require(tx.gasprice == 0, "L2:gasPrice");
 
         TaikoData.Config memory config = getConfig();
-        LibInvalidTxList.Reason reason = LibInvalidTxList.isTxListInvalid({
+        LibInvalidTxList.assertTxListInvalid({
             config: config,
             encoded: txList,
             hint: hint,
             txIdx: txIdx
         });
-        require(reason != LibInvalidTxList.Reason.OK, "L2:reason");
```

Comment:
i believe this assertion is not unnecessary too, but i'm not sure that is it necessary to check `reason == hint` here?

Reply:
Should we add this assertion? Where does the `hint` come from, the Taiko client itself?

Reply:
yep from a client (with a prover) who want to prove that txList is invalid. To inform the protocol whats the reason why this txList is invalid. But right now seems there is no assertion that `hint` **must** be the actual reason? like if `hint == TX_INVALID_SIG` then the `reason` will also always be `TX_INVALID_SIG` too, but emmm if the actual `reason` is `BLOCK_TOO_MANY_TXS`, and we pass `BINARY_TOO_LARGE` as the `hint`, we still can execute this function successfully?

Reply:
Maybe isTxListInvalid should assert hint == reason at the end of it, rather than relying on the caller to do so, if that is intended behavior.

Reply:
I've create another PR on top of this one: https://github.com/taikoxyz/taiko-mono/pull/13096, @cyberhorsey hopefully it answers your question.